packages/litexa-html/README.md

```diff
@@ -44,7 +44,119 @@ project_dir
 ```
 
-## Adding 'mark' SSML Tags
+## Alexa Web API Interface
+
+Integrating your skill with the Alexa Web API will allow it to both send and receive data
+from your web app.
+
+### Skill Manifest
+
+Before your skill can send and receive data through the Alexa Web API, you will need to
+specify the interface in your skill's manifest. You will need to include
+`ALEXA_PRESENTATION_HTML` in your `skill.*` file to let Alexa know that you are using it.
+For example, your skill manifest might look something like this:
+
+```json
+{
+  "manifest": {
+    "apis": {
+      "custom": {
+        "interfaces": [
+          {
+            "type": "ALEXA_PRESENTATION_HTML"
+          }
+        ]
+  ...
+}
+```
+
+### Directives
+
+With Alexa Web API directives, you can send data to your web app from within your skill.
+
+Use the `directive` keyword in your `*.litexa` files to add these directives to your
+Alexa responses. You can get more information about the usage of the `directive` keyword
+from our
+[Litexa language reference](https://litexa.com/reference/#directive).
+
+#### Start Directive
+
+To start your web app, attach the `Alexa.Presentation.HTML.Start` directive to your skill's response.
+Here's an example of what your Litexa code could look like:
+
+```coffeescript
+launch
+    if HTML.isPresent()
+      directive htmlStartDirective(webAppUrl)
+    ...
+```
+
+And the `#htmlStartDirective()` function would return an object like this, and Litexa will add
+it to your skill's response:
+
+```javascript
+function htmlStartDirective(url) {
+    return {
+        type: "Alexa.Presentation.HTML.Start",
+        configuration: {
+            timeoutInSeconds: 1000
+        },
+        data: {},
+        request: {
+            uri: url,
+            method: "GET",
+            headers: {}
+        },
+        transformers: {}
+    };
+}
+```
+
+#### HandleMessage Directive
+
+To send data to your web app, attach the `Alexa.Presentation.HTML.HandleMessage`
+directive to your skill's response. Here's an example of what your Litexa code could look like:
+
+```coffeescript
+fooState
+  when MyIntent
+    directive htmlHandleMessageDirective(message)
+  ...
+```
+
+And the `#htmlHandleMessageDirective()` function would return an object like the one below, and
+Litexa will add it your skill's response:
+
+```javascript
+function htmlHandleMessageDirective(message) {
+    return {
+        type: "Alexa.Presentation.HTML.HandleMessage",
+        message: message,
+        transformers: {}
+    };
+}
+```
+
+### Events
+
+With Alexa Web API events, your skill can receive & handle data sent from your web app.
+
+#### Message Event
+
+To handle data that is sent from your web app, look for the `Alexa.Presentation.HTML.Message`
+event in your `*.litexa` files like you would for an intent-type event (using the `when` keyword).
+For example:
+
+```coffeescript
+barState
+    when Alexa.Presentation.HTML.Message
+      say "I received a message from the web app!"
```

Comment:
Missing the critical part of using the `$request.message` payload. Could add a short intuitive switch statement example.